### PR TITLE
Use native CSPRNG if available

### PR DESCRIPTION
**Pull request by LukasReschke**
Unfortunately only PHP 7…

### CODE REVIEW

File: lib/private/security/securerandom.php [hunk 1]
```diff
@@ -77,16 +77,29 @@ public function getMediumStrengthGenerator() {
...
 	 * @param int $length The length of the generated string
-	 * @param string $characters An optional list of characters to use if no characterlist is
+	 * @param string $characters An optional list of characters to use if no character list is
 	 * 							specified all valid base64 characters are used.
 	 * @return string
 	 * @throws \Exception If the generator is not initialized.
 	 */
-	public function generate($length, $characters = '') {
+	public function generate($length,
+							 $characters = 'ABCDEFGHIJKLMNOPQRSTUVWXYZabcdefghijklmnopqrstuvwxyz0123456789+/') {
```
Comment (DeepDiver1975) [inline lib/private/security/securerandom.php]: indent

Comment (LukasReschke) [inline lib/private/security/securerandom.php]: https://github.com/owncloud/core/pull/20373/files?ts=4 :wink:

File: lib/public/security/isecurerandom.php [hunk 2]
```diff
@@ -70,11 +70,12 @@ public function getMediumStrengthGenerator();
 	/**
 	 * Generate a random string of specified length.
 	 * @param int $length The length of the generated string
-	 * @param string $characters An optional list of characters to use if no characterlist is
+	 * @param string $characters An optional list of characters to use if no character list is
 	 * 							specified all valid base64 characters are used.
 	 * @return string
 	 * @throws \Exception If the generator is not initialized.
-	 * @since 8.0.0
```
Comment (LukasReschke) [inline lib/public/security/isecurerandom.php]: :speak_no_evil: - Should not have removed that. Bloody copy-pasta :wink:

**Comment by MorrisJobke** [comment]
:+1:

**Comment by MorrisJobke** [comment]
OCS API failure unrelated?

**Comment by LukasReschke** [comment]
> OCS API failure unrelated?

Yes.

@DeepDiver1975 Merge-bot requested.

**Comment by DeepDiver1975** [comment]
:+1:

**Pull request merged**